src/config.c

```diff
@@ -829,6 +829,7 @@ void configSetCommand(client *c) {
             goto err;
         }
     }
+    moduleFireServerEvent(REDISMODULE_EVENT_CONFIG_CHANGE, 0, NULL);
```

Comment:
so was thinking about this, the reason I didn't do it was mostly because I didn't understand the memory ownership semantics of it.  (i.e. who owns/frees the struct one is allocating)

for our purposes, the way I'd use my current implementation would be that the module would cache the entries it cares about and if there was a change, it would just compare the current value against the cached value to see if anyone it cared about changed.  but I do see the value in making it better.

in general, I don't think its a good idea to fire it multiple times.  even if its "easier" to implement.

Reply:
@oranagra extended the original PR to include data in the event that enumerates all the config names that were set.